aignostic/aignostic/metrics/metrics.py

```diff
@@ -156,10 +167,58 @@ def macro_recall(name, info: CalculateRequest) -> float:
         ) / len(np.unique(info.true_labels))
 
 
+def _prepare_datasets(info: CalculateRequest):
+    true_labels = np.array(info.true_labels).flatten()
+    predicted_labels = np.array(info.predicted_labels).flatten()
+
+    if not hasattr(info, "protected_attr") or info.protected_attr is None:
```

Comment:
As codecov says, need to add a test that we get this error if input is such that this holds